modules/srv6/cli.c

```diff
@@ -0,0 +1,286 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2025 Olivier Gournet
+
+#include <gr_api.h>
+#include <gr_cli.h>
+#include <gr_srv6_api.h>
+#include <gr_net_types.h>
+
+#include <ecoli.h>
+
+#include <errno.h>
+
+
+static cmd_status_t srv6_steer_add(const struct gr_api_client *c, const struct ec_pnode *p) {
+	struct gr_srv6_steer_add_req *req;
+	const struct ec_pnode *n;
+	const struct ec_strvec *v;
+	const char *str;
+	void *resp_ptr = NULL;
+	size_t len;
+	int ret, i;
+
+	// get NEXT sequence node. it is the parent of the first NEXT node.
+	n = ec_pnode_find(p, "NEXT");
+	if (n == NULL || (n = ec_pnode_get_parent(n)) == NULL || ec_pnode_len(n) < 1)
+		return CMD_ERROR;
+	len = sizeof(*req) + sizeof(req->s.nh[0]) * ec_pnode_len(n);
+	req = calloc(1, len);
+	req->s.n_nh = ec_pnode_len(n);
+
+	// parse NEXT list.
+	for (n = ec_pnode_get_first_child(n), i = 0; n != NULL;	n = ec_pnode_next(n), i++) {
+		v = ec_pnode_get_strvec(n);
+		str = ec_strvec_val(v, 0);
+		if (inet_pton(AF_INET6, str, &req->s.nh[i]) != 1) {
+			free(req);
+			return CMD_ERROR;
+		}
+	}
+
+	if ((str = arg_str(p, "DEST6")) != NULL) {
+		if (ip6_net_parse(str, &req->s.dest6, true) < 0)
+			return CMD_ERROR;
```

Comment:
don't you need a `free(req)` for both error cases DEST6 and DEST4 here after ?

I feel that all the cases:
```
free(req);
return CMD_ERROR;
```

should be replaced by a `goto cmd_error;`

then the end of this function becomes:
```
free(req);
free(resp_ptr);
return CMD_SUCCESS;

cmd_error:
  free(req);
  return CMD_ERROR;
```